content/en/hosting/4.x/upgrade-troubleshooting.md

```diff
@@ -0,0 +1,153 @@
+---
+title: "Troubleshooting 4.x upgrades"
+linkTitle: "Troubleshooting upgrades"
+weight: 50
+aliases:
+  - 
+description: >
+  What to do when CHT 4.x upgrades don't work as planned
+relatedContent: >
+  hosting/4.x/data-migration
+---
+
+4.0.0 was released in November of 2022 so 4.x is mature and users have learned a number of important lessons on how to fix failed 4.x upgrades.  Below are some specific tips as well as general practices on upgrading 4.x.
+
+There's a concept of upgrades "getting stuck" which mainly means that after many many hours an upgrade is not making any progress.  Most likely, this will manifest as the progress bars in the  upgrade admin web UI not increasing and "sticking" at a certain percentage.
```

Comment:
This can also mean the progress bars disappearing altogether.

Reply:
gotcha! fixed in 6fcbd395